platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/SyncInputStream.java

```diff
@@ -18,59 +18,67 @@
 
 import static com.swirlds.common.io.extendable.ExtendableInputStream.extendInputStream;
 
-import com.swirlds.common.crypto.DigestType;
+import com.swirlds.common.config.SocketConfig;
+import com.swirlds.common.context.PlatformContext;
 import com.swirlds.common.crypto.Hash;
 import com.swirlds.common.io.extendable.extensions.CountingStreamExtension;
-import com.swirlds.common.io.extendable.extensions.HashingStreamExtension;
 import com.swirlds.common.io.streams.SerializableDataInputStream;
 import com.swirlds.platform.event.GossipEvent;
 import com.swirlds.platform.gossip.SyncException;
 import com.swirlds.platform.gossip.shadowgraph.Generations;
 import com.swirlds.platform.network.ByteConstants;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.BufferedInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.List;
+import java.util.zip.Inflater;
+import java.util.zip.InflaterInputStream;
 
 public class SyncInputStream extends SerializableDataInputStream {
 
     /** The maximum number of tips allowed per node. */
     private static final int MAX_TIPS_PER_NODE = 1000;
 
     private final CountingStreamExtension syncByteCounter;
-    private final HashingStreamExtension hasher;
 
-    private SyncInputStream(InputStream in, CountingStreamExtension syncByteCounter, HashingStreamExtension hasher) {
+    private SyncInputStream(InputStream in, CountingStreamExtension syncByteCounter) {
         super(in);
         this.syncByteCounter = syncByteCounter;
-        this.hasher = hasher;
     }
 
-    public static SyncInputStream createSyncInputStream(InputStream in, int bufferSize) {
-        CountingStreamExtension syncCounter = new CountingStreamExtension();
-        HashingStreamExtension hasher = new HashingStreamExtension(DigestType.SHA_384);
+    public static SyncInputStream createSyncInputStream(
+            @NonNull final PlatformContext platformContext, @NonNull final InputStream in, final int bufferSize) {
 
-        // the buffered reader reads data first, for efficiency
-        return new SyncInputStream(
-                extendInputStream(new BufferedInputStream(in, bufferSize), syncCounter, hasher), syncCounter, hasher);
+        final CountingStreamExtension syncCounter = new CountingStreamExtension();
+
+        final boolean compress = platformContext
+                .getConfiguration()
+                .getConfigData(SocketConfig.class)
+                .gzipCompression();
+
+        final InputStream meteredStream = extendInputStream(in, syncCounter);
+
+        final InputStream wrappedStream;
+        if (compress) {
```

Comment:
Everyone on the network has to be configured the same way?  Either everyone uses gzip compression or everyone does not?   Is there any value in handshaking or negotiating on connection to use gzip?

Reply:
Some settings can theoretically be varied from node to node, others can not. For example, if the rounds until ancient setting is not the same on a node it will ISS.

For this particular configuration, I don't think we need to worry about supporting varied configuration. If compression improves the performance of the network as a whole, I'd prefer not to give nodes the option of ignoring it.